File: cameraview/src/main/java/com/base/cameraview/engine/action/ActionCallback.java
package com.base.cameraview.engine.action;

import androidx.annotation.NonNull;

/**
 * A callback for {@link Action} state changes.
 * See the action class.
 * <p>
 * See also {@link CompletionCallback}.
 */
public interface ActionCallback {

    /**
     * Action state has just changed.
     *
     * @param action action
     * @param state  new state
     */
    void onActionStateChanged(@NonNull Action action, int state);
}

File: cameraview/src/main/java/com/base/cameraview/engine/action/ActionCallback.kt
package com.base.cameraview.engine.action

/**
 * A callback for [Action] state changes.
 * See the action class.
 *
 *
 * See also [CompletionCallback].
 */
interface ActionCallback {
    /**
     * Action state has just changed.
     *
     * @param action action
     * @param state  new state
     */
    fun onActionStateChanged(action: Action, state: Int)
}
